From 5cc7d46bd05cfd863c3e8765b908877702e9940a Mon Sep 17 00:00:00 2001
From: Ivan Maidanski <ivmai@mail.ru>
Date: Sat, 25 Oct 2014 20:58:06 +0400
Subject: [PATCH 09/20] Fix missing output folder on making auto-generated test
 files (Automake)

* tests/Makefile.am (test_atomic_include.h, list_atomic.c,
list_atomic.i): Add mkdir for output folder.
---
 tests/Makefile.am | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tests/Makefile.am b/tests/Makefile.am
index 6bfe33c..c8812bb 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -35,6 +35,7 @@ test_malloc_LDADD = $(THREADDLLIBS) \
         $(top_builddir)/src/libatomic_ops.la
 
 test_atomic_include.h: test_atomic_include.template
+	mkdir -p `dirname $@`
 	sed -e s:XX::g $? > $@
 	sed -e s:XX:_release:g $? >> $@
 	sed -e s:XX:_acquire:g $? >> $@
@@ -45,6 +46,7 @@ test_atomic_include.h: test_atomic_include.template
 	sed -e s:XX:_acquire_read:g $? >> $@
 
 list_atomic.c: list_atomic.template
+	mkdir -p `dirname $@`
 	echo "#include \"atomic_ops.h\"" > $@
 	sed -e s:XSIZE_::g -e s:XCTYPE:AO_t:g -e s:XX::g $? >> $@
 	sed -e s:XSIZE_::g -e s:XCTYPE:AO_t:g -e s:XX:_release:g $? >> $@
@@ -93,6 +95,7 @@ list_atomic.c: list_atomic.template
 	sed -e s:XSIZE:double:g -e s:XCTYPE:AO_double_t:g -e s:XX:_dd_acquire_read:g $? >> $@
 
 list_atomic.i: list_atomic.c
+	mkdir -p `dirname $@`
 	$(COMPILE) $? -E > $@
 
 # Verify list_atomic.c syntax:
-- 
1.9.3

